src/commands/utils/versioncmp.cc

```diff
@@ -78,5 +78,70 @@ int VersionCompareCmd::execute( Zypper &zypper, const std::vector<std::string> &
   else
     zypper.out().info( str::form(_("%s is older than %s"), lhs.asString().c_str(), rhs.asString().c_str() ) );
 
+  if ( zypper.out().verbosity() > Out::NORMAL ) {
+    auto chclass = [] ( char ch )->unsigned {
+      if ( ch >= '0' && ch <= '9' )
+        return 1;
+      if ( (ch >= 'A' && ch <= 'Z') || (ch >= 'a' && ch <= 'z') )
+        return 2;
+      if ( ch == '~' )
+        return 3;
+      if ( ch == '^' )
+        return 4;
+      return 0;
+    };
+    auto split = [&] ( const std::string & v, std::vector<std::string_view> & vs, std::string_view first ) {
+      vs.clear();
+      vs.push_back( first );
+      for ( const char * s = v.c_str(); *s; ) {
+        unsigned cs = chclass( *s );
+        if ( cs == 0 ) {
+          ++s;
+          continue;
+        }
+        const char * e = s+1;
+        while ( chclass( *e ) == cs )
```

Comment:
If think you change to add a `*e &&` to the while condition, otherwise you'll overrun the buffer if the version ends with a character of class 0.

Reply:
Oh wait, ignore me ;-)